metriq_gym/benchmarks/mirror_circuits.py

```diff
@@ -0,0 +1,274 @@
+
+"""Mirror circuits benchmark for the Metriq Gym.
+
+This benchmark evaluates a quantum device's ability to execute mirror circuits,
+which are quantum circuits with a reflection structure that perform calculations
+and then reverse them. Mirror circuits provide scalable benchmarking capabilities
+for quantum computers as defined in Proctor et al., Nature Physics 2022.
+"""
+
+from dataclasses import dataclass
+from typing import Optional, Tuple
```

Comment:
[nit]: Can we opt for using native types? So `tuple` over `Tuple` and doing things like `int | None` over `Optional[int]`.